bugstalker: init at 0.1.4

jacg opened the pull request:
## Description of changes

Add new package `bugstalker`, a Rust debugger for Linux x86-64: https://github.com/godzie44/BugStalker

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jacg committed:
bugstalker: init at 0.1.4

Aleksanaa commented on pkgs/by-name/bu/bugstalker/package.nix:
```diff
@@ -0,0 +1,35 @@
+
+  nativeBuildInputs = [ pkg-config ];
+
+  doCheck = false;
```
Please add a comment explaining why to disable tests.

If some tests can be enabled, use

```
  checkFlags = [
    # reason for disabling test
    "--skip=example::tests:example_test"
  ];
```

jacg replied:
Many tests fail in the sandbox, even though they pass without a problem in a normal development environment. I haven't managed to work out how to inspect what is going on in the sandbox interactively, so the iteration time of almost 3 minutes makes it prohibitively expensive to get much more insight into it.

The errors all amount to "file not found" without giving any hint as to what file was not found. I suspect they are source files in the `examples` directory. Without an interactive means of working in the sandbox, I haven't managed to figure out how this might be remedied.

jacg committed:
Enable checks which don't fail in the sandbox

66 tests which pass in a normal development environment fail in the
sandbox because of "file not found". I think that this is because they
are trying to find the source code of the examples, but I haven't found
a way of investigating this interactively in the sandbox, it's very
difficult to get to the bottom of it.

So, for now, I have simply disabled the tests that fail in the sandbox.
